src/code/snippets/src/rendering/render_pass.rs

```diff
@@ -48,19 +50,21 @@ impl MyRenderPass {
                 }
             ";
 
-        let shader = GpuProgram::from_source(&renderer.state, "MyShader", vs, fs)?;
+        let server = renderer
+            .server
+            .as_any()
+            .downcast_ref::<GlGraphicsServer>()
+            .unwrap();
+        let shader = GlProgram::from_source(server, "MyShader", vs, fs)?;
 
         Ok(Self {
             enabled: true,
-            world_view_proj: shader.uniform_location(
-                &renderer.state,
-                &ImmutableString::new("worldViewProjectionMatrix"),
-            )?,
```

Comment:
This one requires a uniform buffer. It can be created using `server.create_buffer(256, BufferKind::Uniform, BufferUsage::DynamicDraw)`. Then it should be filled before rendering using  `StaticUniformBuffer::<256>::new().with(Matrix4::identity())` and then it should be registered in the `resource_binding_groups `